src/common/logger.hpp

```diff
@@ -2,24 +2,47 @@
 
 #include <string>
 #include <cstring>
+#include <map>
 #include <memory>
 
 #include "disable_warnings_push.hpp"
 #include "spdlog/logger.h"
 #include "spdlog/spdlog.h"
 #include "disable_warnings_pop.hpp"
 
+namespace {
+
+  spdlog::level::level_enum get_loglvl(const std::string& loglvl) {
+    
+    std::map<std::string, spdlog::level::level_enum> loglvls {
+      {"trace", spdlog::level::trace},
+      {"debug", spdlog::level::debug},
+      {"info" , spdlog::level::info},
+      {"warn" , spdlog::level::warn},
+      {"err"  , spdlog::level::critical},
+      {"off"  , spdlog::level::off}
+    };
+
+    if (auto it = loglvls.find(loglvl); it != loglvls.end()) {
+      return it->second;
+    }
+    else {
+      return spdlog::level::trace;
```

Comment:
This function should not just silently default to trace. It should either raise exception or return `nullopt`.